guest/filter/filter.go

```diff
@@ -115,3 +116,57 @@ func (n *nodeInfo) lazyNode() proto.Node {
 	n.node = &internalproto.Node{Msg: &msg}
 	return n.node
 }
+
+type PodInfo struct {
+	pod proto.Pod
+}
+
+func (p *PodInfo) GetApiVersion() string {
+	return p.lazyPod().GetApiVersion()
+}
+
+func (p *PodInfo) GetKind() string {
+	return p.lazyPod().GetKind()
+}
+
+func (p *PodInfo) GetName() string {
+	return p.lazyPod().GetName()
+}
+
+func (p *PodInfo) GetNamespace() string {
+	return p.lazyPod().GetNamespace()
+}
+
+func (p *PodInfo) GetResourceVersion() string {
+	return p.lazyPod().GetNamespace()
+}
+
+func (p *PodInfo) GetUid() string {
+	return p.lazyPod().GetUid()
+}
+
+func (p *PodInfo) Pod() proto.Pod {
+	return p.lazyPod()
+}
+
+func (p *PodInfo) Spec() *protoapi.PodSpec {
+	return p.lazyPod().Spec()
+}
+
+func (p *PodInfo) Status() *protoapi.PodStatus {
+	return p.lazyPod().Status()
+}
+
+// lazyNode lazy initializes node from imports.Node.
+func (p *PodInfo) lazyPod() proto.Pod {
+	if pod := p.pod; pod != nil {
+		return pod
+	}
+
+	var msg protoapi.Pod
+	if err := imports.Pod(msg.UnmarshalVT); err != nil {
+		panic(err.Error())
+	}
+	p.pod = internalprefilter.Pod
```

Comment:
@sanposhiho
I am not sure it is correct.
If internalprefilter.Pod is used here, `Pod` is used instead of `PodInfoToAdd`, right ???
Or do I misunderstand??

Reply:
Yeah, seemingly it's incorrect.

So, `internalprefilter.Pod` is the Pod being scheduled right now. And, `PodInfoToAdd` is completely another Pod(s).

@chansuke you can implement `internalproto.Pod` like lazyNode above does.